nni/nas/oneshot/pytorch/supermodule/sampling.py

```diff
@@ -76,6 +77,22 @@ def reduction(self, items: list[Any], sampled: list[Any]):
         """Override this to implement customized reduction."""
         return weighted_sum(items)
 
+    def _save_to_sub_state_dict(self, destination, prefix, keep_vars):
+        params_mapping = self._slice_params_mapping()
+        for name, value in itertools.chain(self._parameters.items(), self._buffers.items()):  # direct children
```

Comment:
Maybe you can have a utility function for this, as this for-loop has appeared many times.

Reply:
We split `_save_to_sub_state_dict` into two function `_save_param_buff_to_state_dict`, which save the params and buffers of current module to sate dict, and `_save_module_to_state_dict` which  save the sub module state dict.